### PR TITLE
Fix handling of spaces in `$CRATE_HOME`

### PR DESCRIPTION
**Pull request by mechanomi**
Included changes:

- Drops the unnecessary and potentially misleading hashbang from `crate.in.sh`
  because the file is meant to be sourced not executed.

- Puts `$CRATE_HOME` in double quotes when used in a for loop. This prevents sh
  from interpreting a space in this variable as an item terminator.

- Wraps `${LOGGC}` in nested double quotes. This prevents sh from interpreting
  a space in this variable as an argument terminator when used in `$JAVA_OPTS`.

- Sets `-e` on the `bin/crate` hashbang, ensuring the script exits immediately
  if an error is encountered. This prevents cascading errors.

- Wraps the variables passed to `exec` in two sets of double quotes. This
  preserves one set of double quotes in the subcommand being executed, which is
  necessary for paths with a space in them.

## Summary of the changes / Why this improves CrateDB

right now, if you follow https://crate.io/docs/crate/tutorials/en/latest/install-run/macos.html with pwd that includes spaces (e.g., `/Users/foo/Crate Sandbox`), the user will experience cascading errors when the `try` script attempts to execute `bin/crate`

```
crate-4.3.2/bin/crate: line 129: /Users/foo/Crate: No such file or directory
crate-4.3.2/bin/crate: line 129: exec: /Users/foo/Crate: cannot execute: No such file or directory
```

running `bin/crate` manually produces the same result

the changes in this commit fix this issue on my local computer. I'm not sure if the changes will work on Windows, but am hoping the CI checks will tell me that :D

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [ ] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/appendices/release-notes/unreleased.rst [hunk 1]
```diff
@@ -94,3 +94,5 @@ Fixes
 
 - Fixed a regression introduced in CrateDB ``4.0`` preventing the global setting
   ``cluster.info.update.interval`` to be changed.
+
+- Fixed handling of spaces in `$CRATE_HOME`.
```
Comment (mfussenegger) [inline docs/appendices/release-notes/unreleased.rst:98]: We usually try to include a symptom that users may have seen when running into the fixed issue. 

```suggestion
- Fixed handling of spaces in `$CRATE_HOME`. Users would get a `No such file or directory` error if the path set via `$CRATE_HOME` contained spaces.
```

**Review by mfussenegger — COMMENTED** [review]
Looks like this breaks some tests because of incorrect arguments to java:

https://jenkins.crate.io/blue/organizations/jenkins/CrateDB%2Fcrate_on_pr/detail/crate_on_pr/12420/pipeline#step-123-log-663

```
> Task :blackbox:monitoringTest FAILED
Usage: java [options] <mainclass> [args...]
           (to execute a class)

   or  java [options] -jar <jarfile> [args...]
           (to execute a jar file)

   or  java [options] -m <module>[/<mainclass>] [args...]
       java [options] --module <module>[/<mainclass>] [args...]
           (to execute the main class in a module)

   or  java [options] <sourcefile> [args]
           (to execute a single source-file program)
```

File: docs/appendices/release-notes/unreleased.rst (hunk 1 above)
Comment (mechanomi) [reply docs/appendices/release-notes/unreleased.rst:98]: thx!

**Comment by mechanomi** [comment]
@mfussenegger I have pushed some fixups. this should make the tests pass. I also verified that this new commit fixes the original issue as reported

the culprit was my attempt to use double quotes around every `CLASS_PATH` path. it turns out that works in some instances, but not when the Python tests setup tries to start a CrateDB server. I have now instead fixed the use of spaces in the `CLASS_PATH` by making sure that every time `CLASS_PATH` is used, it is properly wrapped per the first suggestion here: https://stackoverflow.com/a/3003908

**Review by mfussenegger — APPROVED** [review]

**Pull request merged**